### PR TITLE
chore!: update styled-components to 6 & update DS to 2.0.0-beta.3

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* updates styled-components to v6
* integrates changes from design-system 2.0.0-beta.3

### Related issue(s)/PR(s)

* resolves DX-1398
* resolves DX-1253

### CODE REVIEW

**Review by innerdvations — CHANGES_REQUESTED** [review]
Mostly just little things.

The only big thing is my concern about requiring everything to be wrapped in Fields with an explicit label, it feels like the increased verbosity will lead to more mistakes and a steeper learning curve. But this isn't my scope so I won't object too strongly 😆

File: packages/core/admin/admin/src/components/ContentBox.tsx [hunk 1]
```diff
@@ -40,7 +46,7 @@ const ContentBox = ({
   );
 };
 
-const IconWrapper = styled(Flex)`
+const IconWrapper = styled<FlexComponent>(Flex)`
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/core/admin/admin/src/components/ContentBox.tsx:49]: It's an issue with styled components and components that use generics. You'd need to fix it on the library level, which at the moment I'm not prepared to contribute too 😂

**Comment by joshuaellis** [comment]
> Mostly just little things.
> 
> The only big thing is my concern about requiring everything to be wrapped in Fields with an explicit label, it feels like the increased verbosity will lead to more mistakes and a steeper learning curve. But this isn't my scope so I won't object too strongly 😆

Maybe, in an ideal world we'd all just use the form input renderer and we just have one set of fields to maintain, then there is no learning curve.

File: packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/PublishAction.tsx
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/PublishAction.tsx:1]: yeah we can ignore that, @madhurisandbhor is working on re-implementing it all in a different PR.

File: packages/core/content-type-builder/admin/src/components/IconPicker/IconPicker.tsx
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/core/content-type-builder/admin/src/components/IconPicker/IconPicker.tsx:1]: Depends on the context but most likely yes, we typically pass prop as camelCase and convert to kebab-case. But also, we also have bad naming on some props...

File: packages/core/upload/admin/src/components/AssetDialog/BrowseStep/PaginationFooter/components.jsx [hunk 2]
```diff
@@ -14,18 +14,18 @@ const PaginationText = styled(Typography)`
 const LinkWrapper = styled.button`
   padding: ${({ theme }) => theme.spaces[3]};
   border-radius: ${({ theme }) => theme.borderRadius};
-  box-shadow: ${({ active, theme }) => (active ? theme.shadows.filterShadow : undefined)};
+  box-shadow: ${({ $active, theme }) => ($active ? theme.shadows.filterShadow : undefined)};
   text-decoration: none;
   display: flex;
 
-  ${buttonFocusStyle}
+  // TODO: re-add focus style
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/core/upload/admin/src/components/AssetDialog/BrowseStep/PaginationFooter/components.jsx:21]: I forgot this!

**Review by innerdvations — APPROVED** [review]
LGTM!

QAd it and only one small issue now, the focus-visible state for text inputs shows the border on the actual inner text input rather than the outer element.

image

**Comment by joshuaellis** [comment]
> QAd it and only one small issue now, the focus-visible state for text inputs shows the border on the actual inner text input rather than the outer element.

I'll make this a bug in JIRA

**Pull request merged**